liquibase-couchbase/src/main/java/liquibase/ext/couchbase/statement/RemoveDocumentsQueryStatement.java

```diff
@@ -0,0 +1,40 @@
+package liquibase.ext.couchbase.statement;
+
+import com.couchbase.client.java.transactions.ReactiveTransactionAttemptContext;
+import com.couchbase.client.java.transactions.TransactionAttemptContext;
+import liquibase.ext.couchbase.operator.ClusterOperator;
+import liquibase.ext.couchbase.types.Id;
+import liquibase.ext.couchbase.types.Keyspace;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.reactivestreams.Publisher;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Getter
+@EqualsAndHashCode(callSuper = true)
+public class RemoveDocumentsQueryStatement extends RemoveDocumentsStatement {
+
+    private final String whereCondition;
+
+    public RemoveDocumentsQueryStatement(Keyspace keyspace, List<Id> ids, String whereCondition) {
+        super(keyspace, ids);
+        this.whereCondition = whereCondition;
+    }
+
+    @Override
+    public void doInTransaction(TransactionAttemptContext transaction, ClusterOperator clusterOperator) {
+        List<String> filteredIds = clusterOperator.retrieveDocumentIdsByWhereClause(getKeyspace(), whereCondition);
+        getIds().addAll(filteredIds.stream().map(Id::new).collect(Collectors.toList()));
```

Comment:
Maybe we should convert all this to set to remove duplicate ids ?
Imagine that we pass ids and whereClause in changeset. And we already have id "key1" in ids list. And where query also found document with id "key1" and added into list. We will have 2 duplicate ids in the end.

Reply:
Agree, changed IDs container from List to Set